src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java

```diff
@@ -145,6 +145,7 @@ public void informTest() throws Exception {
         assertThat(components).hasSize(1);
 
         final Component component = components.get(0);
+        assertThat(component.getSupplier()).isEqualTo("Example Incorporated"); /*Issue #2373, #2737 */
```

Comment:
`getSupplier()` will return an object of type `OrganizationalEntity`, which you cannot compare with a string.

It should be something along the lines of:

```java
assertThat(component.getSupplier().getName()).isEqualTo("Example Incorporated");
assertThat(component.getSupplier().getContacts().get(0).getName()).isEqualTo("Foo");
assertThat(component.getSupplier().getContacts().get(0).getEmail()).isEqualTo("foo@bar.com");
// etc.
```

For supplier data to be present here, it needs to be added to the BOM this test is using first: https://github.com/DependencyTrack/dependency-track/blob/master/src/test/resources/bom-1.xml

Reply:
I added all fields for manufacture and supplier in this.  Committing shortly :)